sirfilesystem.c

```diff
@@ -218,6 +218,19 @@ char* _sir_getappfilename(void) {
             resolved = false;
             break;
         }
+# elif defined(__HAIKU__)
+        status_t ret = find_path(B_APP_IMAGE_SYMBOL, B_FIND_PATH_IMAGE_PATH, NULL, buffer, PATH_MAX);
```

Comment:
on a technicality, this should be SIR_MAXPATH, which should be set in sirplatform.h, but if it requires more #ifdefs, then there's no harm. I believe by default it should be set to PATH_MAX outside of windows.

Reply:
It's set in a !__WIN__ block; so using SIR_MAXPATH is fine.